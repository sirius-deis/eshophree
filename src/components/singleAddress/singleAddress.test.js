import { render, screen, fireEvent } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import SingleAddress from "./singleAddress";

const mockStore = configureStore([]);

const address = {
  id: 1,
  name: "John Smith",
  title: "Billing Address",
  address: "123 Main St",
  city: "New York",
  state: "NY",
  zipCode: "10001",
  phone: "123-456-7890",
  email: "john@example.com",
};

describe("SingleAddress component", () => {
  let store;

  beforeEach(() => {
    store = mockStore({});
    store.dispatch = jest.fn();
  });
  it("should match snapshot", () => {
    const { container } = render(<Provider store={store}>
      <SingleAddress {...address} />
    </Provider>);
    expect(container).toMatchSnapshot();
  });
  it("should render address details", () => {
    render(<Provider store={store}>
      <SingleAddress {...address} />
    </Provider>);
    expect(screen.getByText("John Smith")).toBeInTheDocument();
    expect(screen.getByText("123 Main St")).toBeInTheDocument();
    expect(screen.getByText(/New York/)).toBeInTheDocument();
    expect(screen.getByText(/NY/)).toBeInTheDocument();
    expect(screen.getByText(/10001/)).toBeInTheDocument();
    expect(screen.getByText(/123-456-7890/)).toBeInTheDocument();
    expect(screen.getByText(/john@example\.com/)).toBeInTheDocument();
  })
  it("should show edit form when Edit button is clicked", () => {
    render(<Provider store={store}>
      <SingleAddress {...address} />
    </Provider>);
    fireEvent.click(screen.getByText("Edit"));
    expect(screen.getByPlaceholderText("Street Address")).toHaveValue("123 Main St");
    expect(screen.getByPlaceholderText("City")).toHaveValue("New York");
    expect(screen.getByPlaceholderText("State")).toHaveValue("NY");
    expect(screen.getByPlaceholderText("Zip Code")).toHaveValue("10001");
    expect(screen.getByPlaceholderText("Phone Number")).toHaveValue("123-456-7890");
    expect(screen.getByPlaceholderText("Email Address")).toHaveValue("john@example.com");
  })
  it("should update form fields when edited", () => {
    render(<Provider store={store}>
      <SingleAddress {...address} />
    </Provider>);
    fireEvent.click(screen.getByText("Edit"));
    const addressInput = screen.getByPlaceholderText("Street Address")
    fireEvent.change(addressInput, {target: {value: "456 Main St"}})
    expect(addressInput).toHaveValue("456 Main St")
  })
  it('should cancel editing when Cancel button is clicked', () => {
    render(<Provider store={store}>
      <SingleAddress {...address} />
    </Provider>);
    fireEvent.click(screen.getByText("Edit"));
    const addressInput = screen.getByPlaceholderText("Street Address")
    fireEvent.change(addressInput, {target: {value: "456 Main St"}})
    fireEvent.click(screen.getByText("Cancel"));
    expect(screen.getByText("123 Main St")).toBeInTheDocument();
    expect(screen.queryByPlaceholderText("Street Address")).not.toBeInTheDocument()
  })
});