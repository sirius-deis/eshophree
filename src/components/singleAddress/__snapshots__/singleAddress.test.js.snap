// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SingleAddress component should match snapshot 1`] = `
<div>
  <form
    class="sc-gxYJeL gGLwKK"
  >
    <div
      class="sc-iVCKna dIcsDr"
    >
      <h2
        class="sc-kPTPQs tiUYv"
      >
        Billing Address
      </h2>
      <button
        class="sc-bgqQcB sc-fFGjHI jHWwsb dQaKvT"
        type="button"
      >
        Edit
      </button>
    </div>
    <div
      class="sc-dicizt fpupii"
    />
    <h3>
      John Smith
    </h3>
    <div
      class="sc-kuWgmH bgXavs"
    >
      <p
        class="sc-duSInm cRhGRE"
      >
        123 Main St
      </p>
      <p
        class="sc-duSInm cRhGRE"
      >
        New York
        , 
        NY
         
        10001
      </p>
      <p
        class="sc-duSInm cRhGRE"
      >
        123-456-7890
      </p>
      <p
        class="sc-duSInm cRhGRE"
      >
        john@example.com
      </p>
    </div>
  </form>
</div>
`;
